craft_parts/plugins/dotnet_v2_plugin.py

```diff
@@ -0,0 +1,401 @@
+
+        if not snap_name:
+            return set()
+
+        logger.info(f"Using .NET SDK content snap: {snap_name}")
+
+        build_snaps = set()
+        build_snaps.add(snap_name)
+
+        return build_snaps
+
+    @override
+    def get_build_packages(self) -> set[str]:
+        """Return a set of required packages to install in the build environment."""
+        return set()
+
+    @override
+    def get_build_environment(self) -> dict[str, str]:
+        """Return a dictionary with the environment to use in the build step."""
+        options = cast(DotnetV2PluginProperties, self._options)
+
+        environment = {
+            "DOTNET_NOLOGO": "1",
+        }
+
+        # .NET binary provided by the user
+        if "dotnet-deps" in self._part_info.part_dependencies:
+            return environment
+
+        build_on = self._part_info.project_info.arch_build_on
+        snap_name = self._generate_snap_name(options)
+
+        if build_on in _DEBIAN_ARCH_TO_DOTNET_RID:
+            environment["LD_LIBRARY_PATH"] = (
+                f"/snap/{snap_name}/current/lib/$CRAFT_ARCH_TRIPLET_BUILD_ON:/snap/{snap_name}/current/usr/lib/$CRAFT_ARCH_TRIPLET_BUILD_ON"
+            )
+        else:
+            raise ValueError(
+                f"Unsupported architecture {build_on!r}. Supported architectures are {humanize_list(_DEBIAN_ARCH_TO_DOTNET_RID.keys(), 'and')}."
+            )
+
+        snap_location = f"/snap/{snap_name}/current"
+        dotnet_path = f"{snap_location}/usr/lib/dotnet"
+        environment["PATH"] = f"{dotnet_path}:${{PATH}}"
+
+        logger.info("Using environment:")
+        for key, value in environment.items():
+            logger.info(f"  {key}={value}")
+
+        return environment
+
+    @override
+    def get_pull_commands(self) -> list[str]:
+        """Return a list of commands to run during the pull step."""
+        options = cast(DotnetV2PluginProperties, self._options)
+
+        build_for = self._part_info.project_info.arch_build_for
+        dotnet_rid = _DEBIAN_ARCH_TO_DOTNET_RID.get(build_for)
+
+        if not dotnet_rid:
+            raise ValueError(
+                f"Unsupported architecture {build_for!r}. Supported architectures are {humanize_list(_DEBIAN_ARCH_TO_DOTNET_RID.keys(), 'and')}."
+            )
+
+        # Restore step
+        restore_cmd = self._get_restore_command(dotnet_rid, options)
+
+        return [restore_cmd]
+
+    @override
+    def get_build_commands(self) -> list[str]:
+        """Return a list of commands to run during the build step."""
+        options = cast(DotnetV2PluginProperties, self._options)
+
+        build_for = self._part_info.project_info.arch_build_for
+        dotnet_rid = _DEBIAN_ARCH_TO_DOTNET_RID.get(build_for)
+
+        if not dotnet_rid:
+            raise ValueError(
+                f"Unsupported architecture {build_for!r}. Supported architectures are {humanize_list(_DEBIAN_ARCH_TO_DOTNET_RID.keys(), 'and')}."
+            )
```

Comment:
pedantic: This is the third occurrence of this, so the rule of 3 kicks in.  You could dedupe with something like:

```python
def _get_rid(arch: str) -> str:
   """Get the dotnet rid for an arch.

   :raises ValueError: If the architecture is unsupported.
   """
  if not dotnet_rid:
    raise ValueError(
      f"Unsupported architecture {build_for!r}. Supported architectures are {humanize_list(_DEBIAN_ARCH_TO_DOTNET_RID.keys(), 'and')}."
    )
```